### ISSUE TITLE
Wouldn't your code run 10x faster if translated to C ? Python is cute and easy but interpreted. 

### ISSUE THREAD

**Issue by Rinksrides2013** [ksigWyatt/Embedded_Compressor#6]

**Comment by ksigWyatt** [comment]
That’s correct @Rinksrides2013. This could be completely translated to C in order to achieve more performant code. As far as I understand `.pyc` extensions compile down to “byte code” to be executed in a similar fashion than most other languages. So the performance hit there would be it seems in the compile time rather than the runtime. So, yes technically it wouldn’t be as fast. But that’s another discussion. 

As for actually trying to translate any of this; we needed to be able to use libraries that were already built. Trying to build all of it from scratch in C might have taken a little longer lol.

Now that the Pi 4 is out also the memory issue might not be as big a problem, if this was done also.

I may at some point come back and translate it, I’m not the best at C however so it may be some time.

If you have some suggestions though on how to start that process I’d appreciate it.
